### ISSUE TITLE
`conda env export` fails

### ISSUE THREAD

**Issue by pitrou** [conda/conda#1935]
```
$ conda info
Current conda install:

             platform : linux-64
        conda version : 3.19.0
  conda-build version : 1.18.2
       python version : 3.4.3.final.0
     requests version : 2.9.0
     root environment : /home/antoine/miniconda3  (writable)
  default environment : /home/antoine/35
     envs directories : /home/antoine/miniconda3/envs
        package cache : /home/antoine/miniconda3/pkgs
         channel URLs : https://conda.anaconda.org/t/<TOKEN>/numba/linux-64/
                        https://conda.anaconda.org/t/<TOKEN>/numba/noarch/
                        https://conda.anaconda.org/t/<TOKEN>/antoine/linux-64/
                        https://conda.anaconda.org/t/<TOKEN>/antoine/noarch/
                        https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : /home/antoine/.condarc
    is foreign system : False

$ conda env export
Error: '/' not allowed in environment name: /home/antoine/35
```

**Comment by heather999** [comment]
I have the same issue if I use --prefix when creating a new conda environment.

**Comment by timothymillar** [comment]
I can confirm this issue when environments are created with --prefix.

Also this is a duplicate of [conda/conda-env/#217](https://github.com/conda/conda-env/issues/217)

**Comment by jaccarmac** [comment]
The same issue occurs on Windows if using MSYS shell, even if the environment is in the default location.

**Comment by luizvital** [comment]
Same error in Ubuntu 14.04, with the environment in the default location

```
$ conda info 
Current conda install:

             platform : linux-64
        conda version : 4.1.2
  conda-build version : not installed
       python version : 2.7.11.final.0
     requests version : 2.9.1
     root environment : /home/nando/miniconda2  (writable)
  default environment : /home/nando/miniconda2/envs/geopandas
     envs directories : /home/nando/miniconda2/envs
        package cache : /home/nando/miniconda2/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False

```

**Comment by thibaultdelor** [comment]
In my case I am not using any prefix, still fail...
I am running on Mac (10.11.4)
```
$ conda info
Current conda install:

             platform : osx-64
        conda version : 4.1.2
  conda-build version : not installed
       python version : 3.5.1.final.0
     requests version : 2.9.1
     root environment : /Users/tibodelor/miniconda3  (writable)
  default environment : /Users/tibodelor/miniconda3
     envs directories : /Users/tibodelor/miniconda3/envs
        package cache : /Users/tibodelor/miniconda3/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/osx-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/osx-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False


$ conda create  --name testPy3 python=3
Fetching package metadata .......
Solving package specifications: ..........

Package plan for installation in environment /Users/tibodelor/miniconda3/envs/testPy3:

The following NEW packages will be INSTALLED:

    openssl:    1.0.2h-1     
    pip:        8.1.2-py35_0 
    python:     3.5.1-0      
    readline:   6.2-2        
    setuptools: 23.0.0-py35_0
    sqlite:     3.13.0-0     
    tk:         8.5.18-0     
    wheel:      0.29.0-py35_0
    xz:         5.0.5-1      
    zlib:       1.2.8-3      

Proceed ([y]/n)? 

Linking packages ...
[      COMPLETE      ]|########################################################################################################################################################| 100%
No psutil available.
To proceed, please conda install psutil#
# To activate this environment, use:
# $ source activate testPy3
#
# To deactivate this environment, use:
# $ source deactivate
#

$ source activate testPy3
prepending /Users/tibodelor/miniconda3/envs/testPy3/bin to PATH

$ conda env export 
Error: '/' not allowed in environment name: /Users/tibodelor/miniconda3/envs/testPy3
```

**Comment by rrehders** [comment]
I encountered this on OSX 10.11.5 and conda 4.1.2
Did not specify the prefix either but still encountered the issue.

However, when I issued the command:
_conda env export -n name > name.yml_
I received the appropriate export file with the name I had chosen.

**Comment by Juanlu001** [comment]
Confirmed that `conda env export -n name` works.

**Comment by vatsan** [comment]
Why was this issue closed? The bug report got hijacked by a couple of users who did not use the "prefix" (i.e. their environments were in the standard location). The original issue still persists. 

If you create a virtual environment at a different location using "--prefix" and attempt to run "conda env export" it fails with the said error message:

"CondaValueError: Value error: '/' not allowed in environment name:"

**Comment by timothymillar** [comment]
Agreed this shouldn't be closed, this bug still exists:
(using ubuntu 17.04)

```
Current conda install:

               platform : linux-64
          conda version : 4.3.21
       conda is private : False
      conda-env version : 4.3.21
    conda-build version : not installed
         python version : 3.6.1.final.0
       requests version : 2.12.4
       root environment : /home/tim/Software/miniconda3  (writable)
    default environment : /home/tim/Software/miniconda3
       envs directories : /home/tim/Software/miniconda3/envs
                          /home/tim/.conda/envs
          package cache : /home/tim/Software/miniconda3/pkgs
                          /home/tim/.conda/pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/linux-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/linux-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/linux-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : None
             netrc file : None
           offline mode : False
             user-agent : conda/4.3.21 requests/2.12.4 CPython/3.6.1 Linux/4.10.0-21-generic debian/stretch/sid glibc/2.24    
                UID:GID : 1000:1000

```

Simplified example:

```
$ conda create -p ~/TEST/example python=3
$ source activate /home/tim/TEST/example
$ conda env export

CondaValueError: '/' not allowed in environment name: /home/tim/TEST/example
```

**Comment by vatsan** [comment]
The workaround as of now is to the following (if you are using a non-standard location for the environment).
1) cd to the location containing your environments.
2) source activate <environment name> (you don't need to specify full path since you are in the environment directory).
3) cd to whereever you have your source code repo.
4) "conda env export " (this will work since there is no absolute path to the environment anymore).
5) source deactivate when done.

**Comment by seoninja13** [comment]
vatsan,
what the <source code repo> should look like?
Thanks

**Comment by donoghuc** [comment]
### TL;DR: 
```
# use -p parameter to pass absolute path of \envs directory to conda env export
$ conda env export -p path/to/envs > env_file.yml
```
### Explanation
Assuming you install conda to a "custom" location using -p. 
When you go to export an environment using something like 
```
conda env export > env_file.yml
```
you will likely get a CondaEnvironmentNotFoundError: Could not find environment:

In order to fix this you can do a 
```
conda info --envs
```
to find the location of your conda environments, simply copy the prefix and supply to conda env export using -p parameter. 

Here is an example: 
First I list my environments to get the prefix 
```
(ds_python) cas@ubuntu:~/working_dir/python_r_ds$ conda info --envs
# conda environments:
#
ds_python             *  /home/cas/miniconda/envs/ds_python
root                     /home/cas/miniconda
```
After copying the path I can provide it to conda env export
```
(ds_python) cas@ubuntu:~/working_dir/python_r_ds$ conda env export -p /home/cas/miniconda/envs/ds_python > ds_python_env.yml
```

**Comment by mlgiter** [comment]
On 

`conda export py36 > py36_packages.txt`

I'm getting error 

> Error: Could not locate 'conda-export'

**Comment by adrienrenaud** [comment]
@donoghuc solution is very useful, but... you end up with the prfix to be append at the end to your `yml` file: `prefix: /home/cas/miniconda/envs/ds_python`

**Comment by delmeyer** [comment]
I agree with @vatsan - Issue #1935 should not be closed. @vatsan 's workaround combined with `conda env export -n name` allows me to manually create the `environment.yml` that I need to permit my projects to be reproduced, however, using `conda env export --prefix env_name > environment.yml`, produced a .yml file that took this form:

name: null
channels:
  - conda-forge
  - defaults
prefix: env_$PATH/env_name

Where the prefix being appended to the end of the .yml file as @adrienrenaud mentioned.
